### ISSUE TITLE
Config validator: Support for MAP types

### ISSUE THREAD

**Issue by mikkokar** [HotelsDotCom/styx#143]
## The problem

Config validator is unable to validate `plugins.all` or `services.factories` attributes. This is because those attributes conceptually mappings of strings (or names) to values, but schema DSL only supports rigid record types (object) whose attribute names are all known a priori. In case of map types, the keys are not known a priori.

For this reason `plugins.all` and `services.factories` are declared as opaque types and are not validated.

## Acceptance criteria

 * The `plugins.all` attribute is validated.
 * The `services.factories` attribute is validated.
 * The schema DSL shall have a new *map* type for named mappings.

An example DSL could look like:

```
schema(
    field("factories", mapOf(object(
            field("class", string()),
            field("config", object(opaque()))
        )))
)
```

Note that the "key" type is omitted because for styx configurations this is always String (for now).

**Comment by mikkokar** [comment]
A PR: #148.

**Comment by mikkokar** [comment]
Fixed by #148.
